x/gov/handler.go

```diff
@@ -0,0 +1,76 @@
+package gov
+
+import (
+	"reflect"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Handle all "gov" type messages.
+func NewHandler(keeper Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		switch msg := msg.(type) {
+		case MsgDeposit:
+			return handleMsgDeposit(ctx, keeper, msg)
+		case MsgSubmitProposal:
+			return handleMsgSubmitProposal(ctx, keeper, msg)
+		case MsgVote:
+			return handleMsgVote(ctx, keeper, msg)
+		default:
+			errMsg := "Unrecognized gov Msg type: " + reflect.TypeOf(msg).Name()
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+// Handle MsgSubmitProposal.
```

Comment:
comments like this on unexposed messages are kinda useless I'd just remove them or replace them with a more valuable description